src/arrayMethodSort.js

```diff
@@ -4,9 +4,39 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(compareFunction = compareString) {
+    let count = 1;
+
+    while (count > 0) {
+      count = 0;
+
+      for (let i = 1; i < this.length; i++) {
+        const currentValue = this[i];
+        const prevValue = this[i - 1];
+
+        if (compareFunction(prevValue, currentValue) > 0) {
+          this[i - 1] = currentValue;
+          this[i] = prevValue;
+          count++;
+        }
+      }
+    }
+
+    return this;
   };
 }
 
+const compareString = (prevValue, currentValue) => {
+  const strPrev = `${prevValue}`;
+  const strCurrent = `${currentValue}`;
```

Comment:
Think of different way of type conversion, using interpolation is definitely not the best decision here